core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -133,6 +133,36 @@ angular.module('oppia').controller('RteHelperModalController', [
       $uibModalInstance.dismiss(true);
     };
 
+    $scope.disableSaveButtonForRte = function() {
+      // This method disables the save button for RTE if they violate
+      // the validations defined in rich_text_components_definitions.ts file.
+      let result: boolean = false;
+      $scope.customizationArgSpecs.forEach((arg, index: number) => {
+        let tCA = $scope.tmpCustomizationArgs;
+        if (arg.schema.validators) {
+          let id: string = arg.schema.validators[0].id;
+          if (id.endsWith('least')) {
+            let minValue = arg.schema.validators[0].min_value;
+            if (id.includes('length')) {
+              result = result || tCA[index].value.length < minValue;
+            } else {
+              result = result || tCA[index].value < minValue;
+            }
+          } else if (id.endsWith('most')) {
+            let maxValue = arg.schema.validators[0].max_value;
+            if (id.includes('length')) {
+              result = result || tCA[index].value.length > maxValue;
+            } else {
+              result = result || tCA[index].value > maxValue;
+            }
+          } else if (id.endsWith('nonempty')) {
+            result = result || tCA[index].value.length <= 0;
+          }
```

Comment:
Please avoid using regex and an if-else ladder for this, it is not scalable. Imagine a scenario where a new rule is added, the author will need to remember to make a change here -- not doin so can lead to unexpected bugs in the future. Could you check where RTE component specific validation happen? Perhaps there is a way to relay the output of the validation to this component.

Reply:
@kevintab95 I located that the validation check is taking place inside `apply-validation.directive.ts` file, but I am unable to call its `validate()` function from this `rte-helper-modal.controrller.ts` file ( as I cant understand how to pass the `AbstractControl` instance as argument).
Rather I found that the `apply-validation.directive.ts` file is used via  the `applyValidation` directive in the html tags, which doesnt seem to work in my case.
Nor am I able to implement the `Validator` interface directly from this controller.

Please help with some suitable approach.

Reply:
@kevintab95 PTAL

Reply:
Great, applyValidationDirective is a good find. We use this directive to validate some of our schema based editors. The validate method defined in the directive executes automatically everytime there is a change in the input (please refer [Angular form validation](https://angular.io/guide/form-validation#validating-form-input) for more details). Note that each RTE component will have its own customisation args, so there will be a combination of these schema based editors activated depending on what component is selected. We will need to get the "validity" of all the relevant schema based editors that are active for the component to determine whether to enable / disable the save button. I would suggest looking into how listen to the changes in the validity of the individual inputs -- I think the [form validation documentation](https://angular.io/guide/form-validation#validating-form-input) may be useful.